### PR TITLE
[Meta] Reorganise according to new filename format

### LINKED ISSUE #52: Use kmlmc and kmlmp in the build script
Blocked by:

 - [ ] keymanapp/keyman#1995
 - [ ] keymanapp/keyman#1998

### LINKED ISSUE #53: [Meta] Rename folders from e.g. nrc/str.sencoten to nrc/nrc.str.sencoten
For ease of dealing with folder names, and consistency with the keyboards repo and the downloads.keyman.com folder names, I'd like to rename the model folders to match the model id in its entirety. This simplifies some processes that then don't need to parse the model id in order to derive the folder name.

A cursory look at this suggests that this is a build environment change only, with no other significant side-effects.

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #52.
Fixes #53.
Closes keymanapp/keyman#1973.

1. Rename model files and folders according to #53 -- use full model id in filename and folder name
2. Use kmlmc (and kmlmp) instead of calling node directly (#52)
3. Clean up model files to use new declarative style + export for lexical models.
4. Allow building of example models in sample/ folder (but don't build by default)

TODO:

- [x] Enable use of kmlmp when ready (keymanapp/keyman#2024)
- [x] Add kmlmi to compile lexical model .model_info files (keymanapp/keyman#2024)
- [x] Consider adding .kpj files for each model project (not needed for build, but helps for usage in Keyman Developer)

This PR is not ready for merge until we have addressed the first three TODO items above.

For a future PR:

- [ ] Use npm published versions instead of npm link (no need to compile the lm compiler after that) (keymanapp/keyman#1995)

### CODE REVIEW

File: sample/example/example.en.custom/source/example.en.custom.model.ts [hunk 1]
```diff
@@ -1,6 +1,4 @@
-import LexicalModelCompiler from "@keymanapp/developer-lexical-model-compiler"; 
-
-(new LexicalModelCompiler).compile({
+const source: LexicalModelSource = {
   format: 'custom-1.0',
   wordBreaking: {
     sources: ['wordbreak.ts'],
```
[the remark this answers is not in the document]
Comment (mcdurdin) [reply sample/example/example.en.custom/source/example.en.custom.model.ts]: I think that's best. Good if the samples not only compile, but work :)

File: sample/example/example.crk.wordlist_wahkohtowin/source/example.crk.wordlist_wahkohtowin.model.ts [hunk 2]
```diff
@@ -1,11 +1,11 @@
-import LexicalModelCompiler from "@keymanapp/developer-lexical-model-compiler"; 
-
-(new LexicalModelCompiler).compile({
+const source: LexicalModelSource = {
   format: 'trie-1.0',
   wordBreaking: {
```
[the remark this answers is not in the document]
Comment (mcdurdin) [reply sample/example/example.crk.wordlist_wahkohtowin/source/example.crk.wordlist_wahkohtowin.model.ts]: For now, yes.

File: resources/compile.sh [hunk 3]
```diff
@@ -1,5 +1,8 @@
 #!/bin/bash
 
+set -e
+# set -u We need to do more cleanup to enable this
```
[the remark this answers is not in the document]
Comment (mcdurdin) [reply resources/compile.sh]: (╯°□°）╯︵ ┻━┻

File: resources/compile.sh [hunk 4]
```diff
@@ -133,13 +136,33 @@ function build_release_model {
...
+  local modelInputFilename="$base_model.model.ts"
+  local modelOutputFilename="$base_model.model.js"
+  local modelInputPackageFilename="$base_model.model.kps"
+  local modelOutputPackageFilename="$base_model.model.kmp"
+  local modelInfoFilename="$base_model.model_info"
+
+  pushd source
+
+  # Compile model
+  kmlmc -o "../build/$modelOutputFilename" "./$modelInputFilename" || die "Unable to build .model.js file"
```
Comment (darcywong00) [inline resources/compile.sh]: Would it help the logs to include `$modelInputFilename` if the build fails?
Also for line 151

Comment (mcdurdin) [reply resources/compile.sh]: The model name is clearly logged above so it's not an issue :)

File: sample/example/example.en.custom/source/example.en.custom.model.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jahorton) [reply sample/example/example.en.custom/source/example.en.custom.model.ts]: Which part?  Specifying a .ts file array for the wordBreak function?

[the remark this answers is not in the document]
Comment (eddieantonio) [reply sample/example/example.en.custom/source/example.en.custom.model.ts]: @jahorton Correct! The compiler used to produce junk, syntactically-incorrect code if that was run, and there were no checks implemented to make it run correctly.

File: release/nrc/nrc.str.sencoten/source/nrc.str.sencoten.model.ts [hunk 5]
```diff
@@ -1,8 +1,6 @@
-import LexicalModelCompiler from "@keymanapp/developer-lexical-model-compiler";
-
-(new LexicalModelCompiler).compile({
+const source: LexicalModelSource = {
   format: 'trie-1.0',
-  wordBreaking: 'default',
+  wordBreaker: 'default',
```
Comment (eddieantonio) [inline release/nrc/nrc.str.sencoten/source/nrc.str.sencoten.model.ts]: 👍

File: resources/compile.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (eddieantonio) [reply resources/compile.sh]: Can this be addressed now?

File: resources/compile.sh (hunk 4 above)
Comment (eddieantonio) [inline resources/compile.sh]: Oh yeaaaaah, this is nice! 😍

File: resources/compile.sh [hunk 6]
```diff
@@ -133,13 +136,33 @@ function build_release_model {
...
+
+  # Compile package
+  kmlmp -o "../build/$modelOutputPackageFilename" "./$modelInputPackageFilename" || die "Unable to build .model.kmp file"
+
   popd
 
+  # Merge .model_info file
+
+  kmlmi \
```
Comment (eddieantonio) [inline resources/compile.sh]: I like how obvious this invocation is!

File: sample/example/example.crk.wordlist_wahkohtowin/source/example.crk.wordlist_wahkohtowin.model.ts [hunk 7]
```diff
@@ -1,11 +1,11 @@
-import LexicalModelCompiler from "@keymanapp/developer-lexical-model-compiler"; 
-
-(new LexicalModelCompiler).compile({
+const source: LexicalModelSource = {
   format: 'trie-1.0',
-  wordBreaking: {
+  /*wordBreaker: { //THIS IS THEORETICAL FUTURE FUNCTIONALITY
```
Comment (eddieantonio) [inline sample/example/example.crk.wordlist_wahkohtowin/source/example.crk.wordlist_wahkohtowin.model.ts]: :+1:

File: sample/example/example.en.custom/source/example.en.custom.model.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jahorton) [reply sample/example/example.en.custom/source/example.en.custom.model.ts]: So we'll probably need an issue to resolve this on both ends, then?

[the remark this answers is not in the document]
Comment (eddieantonio) [reply sample/example/example.en.custom/source/example.en.custom.model.ts]: Yes, we'll need a few issues: implement in Keyman; create a valid sample in lexical-models, and document the API and how to actually implement it in help.keyman.com.

**Pull request merged**